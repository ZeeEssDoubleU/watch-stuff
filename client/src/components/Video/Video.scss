   .video-container {
      position: relative;
      padding-top: 9/16 * 100%;
   }

   .video-player {
      position: absolute;
      width: 100%;
      height: 100%;
      top: 0;
   }