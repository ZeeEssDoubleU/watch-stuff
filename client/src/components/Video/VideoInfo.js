import React, { useState } from "react";
import { Button, Image } from "semantic-ui-react";
import Linkify from "react-linkify";

import "./VideoInfo.scss";

import { getFormattedDate } from "../../utils/format-time";
import { getAbbrevNumber } from "../../utils/format-number";

const VideoInfo = props => {
	const [collapsed, toggleCollapse] = useState(true);
	const channel = props.channel;
	const video = props.video;

	if (!video) return <div />;

	const channelIcon =
		"https://yt3.ggpht.com/a-/AAuE7mAWpJP0ftMStYJD1f_tGS-RlvHU2rk7zcxRbg=s240-mo-c-c0xffffffff-rj-k-no";
	const channelName = channel
		? channel.snippet.title
		: video.snippet.channelTitle;
	const publishDate = getFormattedDate(video.snippet.publishedAt);
	const description = video.snippet.description;
	const subCount = channel
		? getAbbrevNumber(channel.statistics.subscriberCount)
		: null;

	return (
		<div className="video-info-container">
			<Image
				className="channel-image"
				src={channelIcon}
				circular
			/>
			<div className="video-info">
				<div className="channel-name">{channelName}</div>
				<div className="videopublication-date">
					Published on {publishDate}
				</div>
			</div>
			<Button className="subscribe-button" color="youtube">
				SUBSCRIBE {subCount}
			</Button>
			<div className="video-description">
				<div className={collapsed ? "collapsed" : "expanded"}>
					<Linkify>{description}</Linkify>
				</div>
				<Button
					className="show-more-button"
					compact
					onClick={() => toggleCollapse(!collapsed)}>
					{collapsed ? "SHOW MORE" : "SHOW LESS"}
				</Button>
			</div>
		</div>
	);
};

export default VideoInfo;
