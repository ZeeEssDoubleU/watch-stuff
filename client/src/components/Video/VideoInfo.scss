@import '../../styles/shared.scss';
.video-info-container {
   display: grid;
   grid: auto auto / max-content auto 156px;
   align-items: center;
   grid-row-gap: 16px;
   .channel-image {
      margin-right: $avatar-margin;
      grid-area: 1/1 / 2/2;
      cursor: pointer;
   }
   .video-info {
      grid-area: 1/2 / 2/3;
      .channel-name {
         font-weight: 600;
         cursor: pointer;
      }
      .video-publication-date {
         font-size: 13px;
         color: #707070;
      }
   }
   .subscribe-button {
      grid-area: 1/3 / 2/4;
      &.ui.button {
         margin: 0;
      }
   }
   .video-description {
      grid-area: 2/2 / 3/4;
      max-width: 615px;
      white-space: pre-line;
      &>.ui.button {
         margin: 8px 0;
         padding-left: 0;
         padding-right: 0;
         background: none;
      }
      p {
         line-height: 2rem;
         margin-bottom: 0;
      }
      .collapsed {
         max-height: 4rem;
         overflow-y: hidden;
      }
      .expanded {
         max-height: none;
      }
   }
}