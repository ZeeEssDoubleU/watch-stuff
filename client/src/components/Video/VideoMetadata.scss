@import "../../styles/shared.scss";

.video-metadata {
	color: #707070;
	font-size: 16px;

	h3 {
		font-weight: 400;
		color: $text-color-dark;
	}

	.video-stats {
		display: flex;
		justify-content: space-between;
		width: 100%;
		align-items: center;

		.video-actions {
			display: flex;
			align-items: center;

			.button {
				height: 30px;

				span {
					margin-left: 5px;
				}

				.highlight {
					color: red;
				}

				.icon {
					transition: transform 0.2s;
				}

				&:hover {
					.icon {
						transform: scale(1.2);
					}
				}
			}

			.ellipsis {
				padding: 0 5px;
				border: none;
				box-shadow: none;
			}

			& > * {
				margin-left: 20px;
				margin-right: 0;
			}
		}
	}
}
