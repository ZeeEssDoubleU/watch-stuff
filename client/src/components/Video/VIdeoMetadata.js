import React from "react";
import { Button, Divider, Icon } from "semantic-ui-react";

import "./VideoMetadata.scss";
import Ratings from "../Ratings/Ratings";
import RatingsBar from "../Ratings/RatingsBar";

const VideoMetadata = props => {
	if (!props.video) return <div />;

	const title = props.video.snippet.title;
	const viewCount = Number(props.video.statistics.viewCount).toLocaleString();
	const likeCount = Number(props.video.statistics.likeCount);
	const dislikeCount = Number(props.video.statistics.dislikeCount);

	return (
		<div className="video-metadata">
			<h3>{title}</h3>
			<div className="video-stats">
				<span>{viewCount} views</span>
				<div className="video-actions">
					<Ratings likes={likeCount} dislikes={dislikeCount} />
					<RatingsBar likes={likeCount} dislikes={dislikeCount} />
					<Button basic labelPosition="left">
						<Icon name="share" />
						<span>SHARE</span>
					</Button>
					<Button basic labelPosition="left">
						<Icon name="add circle" />
						<span>SAVE</span>
					</Button>
					<Button basic className="ellipsis">
						<Icon name="ellipsis horizontal" />
					</Button>
				</div>
			</div>
			<Divider />
		</div>
	);
};

export default VideoMetadata;
