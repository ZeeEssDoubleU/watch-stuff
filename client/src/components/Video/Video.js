import React from "react";
import PropTypes from "prop-types";

import "./Video.scss";

const Video = props => {
	if (!props.id) return null;

	return (
		<div className="video-container">
			<iframe
				className="video-player"
				src={`https://www.youtube.com/embed/${props.id}`}
				frameBorder="0"
				allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
				allowFullScreen
				title={"video"}
			/>
		</div>
	);
};

Video.propTypes = {};

export default Video;
