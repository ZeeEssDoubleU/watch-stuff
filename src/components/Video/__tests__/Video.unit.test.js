import React from "react";
import Video from "../Video";
import { shallow } from "enzyme";

describe("Video", () => {
	test("renders video component correctly", () => {
		const wrapper = shallow(<Video id="vDjcWlCT8rg" />);
		expect(wrapper).toMatchSnapshot();
	});

	test("renders null if id in video component not specified", () => {
		const wrapper = shallow(<Video />);
		expect(wrapper).toMatchSnapshot();
	});
});
