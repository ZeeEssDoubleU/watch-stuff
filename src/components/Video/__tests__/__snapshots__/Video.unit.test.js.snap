// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders null if id in video component not specified 1`] = `""`;

exports[`Video renders video component correctly 1`] = `
<div
  className="video-container"
>
  <iframe
    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    className="video-player"
    frameBorder="0"
    src="https://www.youtube.com/embed/vDjcWlCT8rg"
    title="video"
  />
</div>
`;
